src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -430,29 +430,29 @@ internal void RaiseEvent(string name, string input)
         {
             lock (this.pendingExternalEvents)
             {
-                object tcsObject;
-                if (this.pendingExternalEvents.TryGetValue(name, out tcsObject))
+                ICollection tcsColleciton;
+                if (this.pendingExternalEvents.TryGetValue(name, out tcsColleciton))
                 {
-                    var tcsList = tcsObject as IList;
-                    if (tcsList == null || tcsList.Count == 0)
+                    if (tcsColleciton == null || tcsColleciton.Count == 0)
                     {
                         return;
                     }
 
-                    var tcs = tcsList[0];
+                    Queue tcsQueue = new Queue(tcsColleciton);
+
+                    var tcs = tcsQueue.Dequeue();
                     Type tcsType = tcs.GetType();
                     Type genericTypeArgument = tcsType.GetGenericArguments()[0];
 
                     // If we're going to raise an event we should remove it from the pending collection
                     // because otherwise WaitForExternal() will always find one with this key and run infinitely. Fixes #141
-                    if (tcsList.Count == 1)
+                    if (tcsQueue.Count == 0)
                     {
                         this.pendingExternalEvents.Remove(name);
                     }
                     else
                     {
-                        tcsList.RemoveAt(0);
-                        this.pendingExternalEvents[name] = tcsList;
+                        this.pendingExternalEvents[name] = tcsQueue;

```

Comment:
This entire else block can be removed.

Reply:
Great! This will simplify the code. Thank you!